src/libstd/sys/windows/process.rs

```diff
@@ -141,12 +137,32 @@ impl Command {
                                    .with_extension(env::consts::EXE_EXTENSION);
                     if fs::metadata(&path).is_ok() {
                         return Some(path.into_os_string())
+                    } else {
+                        // Windows relies on path extensions to resolve commands.
+                        // Path extensions are found in the PATHEXT environment variable.
+                        if let Some(exts) = env_lookup(self.env.as_ref(), "PATHEXT") {
```

Comment:
Oh here `self.env.as_ref()` is already bound above as `env` (e.g. we know it's `Some` already). You can change this to just `env.get`, but I was thinking that we'd just want to hoist this out of the loop here so it's only executed once instead of once-per-path.